code/parachain/runtime/picasso/src/lib.rs

```diff
@@ -178,6 +197,68 @@ parameter_types! {
 	pub const SS58Prefix: u8 = 49;
 }
 
+pub struct IbcDenomToAssetIdConversion;
+
+impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
+	type Error = DispatchError;
+
+	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
+		if let Some(id) = IbcDenoms::<Runtime>::get(denom.as_bytes()) {
+			return Ok(id)
+		}
+
+		let asset_id =
+			<currency_factory::Pallet<Runtime> as CurrencyFactoryT>::create(RangeId::IBC_ASSETS)?;
+		let denom_bytes = denom.as_bytes();
+		IbcDenoms::<Runtime>::insert(denom_bytes.clone(), asset_id);
+		IbcAssetIds::<Runtime>::insert(asset_id, denom_bytes);
+
+		let location = XcmAssetLocation::new(MultiLocation::new(
+			1,
+			X1(Junction::GeneralIndex(asset_id.into())),
+		));
+		assets_registry::Pallet::<Runtime>::set_reserve_location(
+			asset_id,
+			location,
+			Rational64::one(),
+			Some(12),
+		)?;
+
+		Ok(asset_id)
+	}
+
+	fn from_asset_id_to_denom(id: CurrencyId) -> Option<String> {
+		IbcAssetIds::<Runtime>::get(id).and_then(|denom| {
+			String::from_utf8(denom)
+				// TODO: replace this for `inspect_err` when it's available as a stable feature?
+				.map_err(|e| {
+					log::error!("Invalid encoding for asset id {} - err = {:?}", id, e);
+					e
+				})
+				.ok()
+		})
+	}
```

Comment:
This can be handled by pallet-assets-registry

Reply:
can you please add that interaction?